src/educational-programs/educational-programs.service.ts

```diff
@@ -16,45 +16,59 @@ export class EducationalProgramsService {
  * @param data Program data to register
  * @returns Registered Educational Program
  */
-  async createProg(data: Prisma.EducationalProgramsCreateInput): Promise<EducationalPrograms> {
-    const existing = await this.prisma.educationalPrograms.findFirst({
-      where:{
-        description: data.description,
-      },
-    });
-    if(existing){
-      throw new BadRequestException(`This educational program already exists`);
-    }
-   return this.prisma.educationalPrograms.create({
-    data,
-   })
+async create(educationalProgram: CreateEducationalProgramDto): Promise <{ message: string | null; error: { message: string } | null; data: EducationalPrograms | null }> {
+  try {
+  await this.validateAreaId(educationalProgram);
+  const newPartialTemplate = await this.prisma.educationalPrograms.create({
+    data: {
+      ...educationalProgram,
+    },
+  });
+
+  return { message: 'Registrado con éxito', error: null, data: newPartialTemplate};
+} catch (error) {
+  return { message: 'No se pudo registrar el programa educativo', error: error.message , data: null };
   }
+ }
 
   /**
    * Method to consult all programs
    * @returns Returns all registered educational programs
    */
-  async findAll(): Promise<EducationalPrograms[]> {
-    return await this.prisma.educationalPrograms.findMany();
+  async findAll(): Promise<any> {
+    try {
+      const programs = await this.prisma.educationalPrograms.findMany();
+      
+      if (programs.length === 0) {
+        return { message: 'Sin programas educativos', error: null, data: null };
+      }
+      
+     
+      return { message: 'Programas educativos encontrados', error: null, data: programs };
+    } catch (error) {
+      return { message: 'Error al cargar programas educativos', error: error.message, data: null };
+    } finally {
+      await this.prisma.$disconnect();
+    }
   }
 
-  /**
-   * Method to query a programs from the id
-   * @param id id of the program to consult
-   * @returns Returns the program with the corresponding id
-   */
 
-  async byId(id: number): Promise<EducationalPrograms> {
-    const edu_project= await this.prisma.educationalPrograms.findUnique({
-      where:{
-        id,
-      },
+   /**
+   * Method to find an educational program by its id.
+   * @param id ID of the program to find
+   * @returns Educational program
+   * @throws NotFoundException if program is not found
+   */
+   async byId(id: number): Promise<EducationalPrograms> {
```

Comment:
update the function name with a name that describes the functionality being performed